File: week1/11000_강의실배정_김태린.py
const fs = require('fs')

let a = fs.readFileSync('input.txt').toString().trim().split('\n')
let list = []
let cnt = 0;

for(let i=1; i<=a[0]; i++) {
  let [b, c] = a[i].split(' ').map(Number)
  list.push([b,c])
}

list.sort((a,b) => a[0] - b[0])

let answer = [];

for(let i=0; i<=list.length; i++) {
  answer.push(list[0])
  list.shift()  
  for(let j=0; j<list.length; j++) {
    let cnt =0
    if(list[j][0] >= answer[answer.length-1][1]) {
      cnt += 1
      if(cnt == 1) {
        answer.push(list[j])
        list.splice(j, 1)
      }
    }
  }
  answer.push('구분')
}

let d= 0
for(let i=0; i<answer.length; i++) {
  if(answer[i] == '구분') {
    d+= 1
  }
}

console.log(d)

File: week1/김태린/11000_강의실배정_김태린.js
const fs = require('fs')

let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

let list = []

for(let i=1; i<=input[0]; i++) {
  let [b, c] = input[i].split(' ').map(Number)
  list.push([b,c])
}

//강의 시간으로 정렬
list.sort((a,b) => a[0] - b[0])

let answer = [];

//가장 첫번째 수업의 종료 시간
answer.push(list[0][1])


for(let i=1; i<list.length; i++) {
  //종료시간이 시작시간보다 크면 배열에 넣기
  if(answer[0] > list[i][0]) {
    answer.push(list[i][1])
  }else {
    answer.shift()
    answer.push(list[i][1])
  }
  answer.sort((a,b) => a-b)
}

console.log(answer.length)